examples/contrib/oed/gp_bayes_opt.py

```diff
@@ -0,0 +1,164 @@
+# import matplotlib.gridspec as gridspec
+# import matplotlib.pyplot as plt
+import torch
+import torch.autograd as autograd
+import torch.optim as optim
+from torch.distributions import transform_to
+
+
+class GPBayesOptimizer:
+    """Performs Bayesian Optimization using a Gaussian Process as an
+    emulator for the unknown function.
```

Comment:
Once the code is a little more settled down, you should expand this docstring and add a couple usage examples